_re2.cc

```diff
@@ -452,9 +452,13 @@ create_regexp(PyObject* self, PyObject* pattern, PyObject* error_class)
   }
 
   if (!regexp->re2_obj->ok()) {
-    long code = (long)regexp->re2_obj->error_code();
     const std::string& msg = regexp->re2_obj->error();
+#if PY_MAJOR_VERSION >= 3
+    PyObject* value = Py_BuildValue("s#", msg.data(), msg.length());
```

Comment:
I think this can be simplified to `PyUnicode_FromStringAndSize`, or use `PyUnicode_FromFormat` to include both the code and the message.  (It looks like we'd have to assume that msg has no embedded null bytes, but that seems reasonable.)

Reply:
Which do you prefer? Err code in or out?

Pros/cons:

Including errcode:
* Lets users write their own error messages (albeit painfully -- they'd need to parse the string)
* Lets users delve into re2 source code a bit more quickly

Nixing errcode:
* Lets users display re2 messages as-is (as in my case -- I'm building user-facing errors and for now, re2's own error strings are good)
* Behaves more like `re.error`.
* Is backwards-compatible with previous versions of pyre2 (I think?)

My vote's for no-errcode. I expect a future version of pyre2 could expose errcode by extending `sre.error` with a subclass that includes errcode -- which seems like the best of all worlds?

But I could be missing something. I haven't been using pyre2 very long, and Python 2 is a distant memory for me. I don't understand where that error code goes in the Python 2 case. I can't see it when I run:

```
pyre2:py3-error *$ python
Python 2.7.15 (default, Oct 15 2018, 15:26:09)
[GCC 8.2.1 20180801 (Red Hat 8.2.1-2)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import re
>>> try: re.compile('*')
... except Exception as e: err = e
...
>>> e.args
('nothing to repeat',)
>>> e.message
'nothing to repeat'
>>> dir(e)
['__class__', '__delattr__', '__dict__', '__doc__', '__format__', '__getattribute__', '__getitem__', '__getslice__', '__hash__', '__init__', '__module__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__setstate__', '__sizeof__', '__str__', '__subclasshook__', '__unicode__', '__weakref__', 'args', 'message']
```

Reply:
Sounds like no error code is the better option.  Maybe the reason you don't see the code in your example is that you imported `re` instead of `re2`?

Reply:
whoop -- indeed, `re2` is not `re` :)

Still, it doesn't look like nixing will cause anybody a serious issue. Pushing the nix.

```
pyre2:py3-error *$ python
Python 2.7.15 (default, Oct 15 2018, 15:26:09)
[GCC 8.2.1 20180801 (Red Hat 8.2.1-2)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import re2
>>> try: re2.compile('*')
... except Exception as err: e = err
...
>>> e.args
(8, 'no argument for repetition operator: *')
>>> e.message
''
>>> str(e)
"(8, 'no argument for repetition operator: *')"
>>> e.msg
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'error' object has no attribute 'msg'
>>> dir(e)
['__class__', '__delattr__', '__dict__', '__doc__', '__format__', '__getattribute__', '__getitem__', '__getslice__', '__hash__', '__init__', '__module__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__setstate__', '__sizeof__', '__str__', '__subclasshook__', '__unicode__', '__weakref__', 'args', 'message']
```